code/modules/mob/living/simple_animal/abnormality/teth/so_that_no_cry.dm

```diff
@@ -50,12 +74,154 @@
+	playsound(src, 'sound/abnormalities/so_that_no_cry/counter.ogg', min(15 + damage, 100), TRUE, 4)
+	attacker.visible_message("<span class='danger'>[src] hits [attacker] with a barrage of punches!</span>", "<span class='userdanger'>[src] counters your attack!</span>")
+	do_attack_animation(attacker)
+	attacker.apply_damage(damage, attack_type, null, attacker.getarmor(null, attack_type))
+	new /obj/effect/temp_visual/revenant(get_turf(attacker))
+
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/Move()
+	if(!can_act)
+		return FALSE
+	return ..()
+
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/attack_hand(mob/living/carbon/human/M)
+	..()
+	if(!.)
+		return
+	if(damage_reflection && M.a_intent == INTENT_HARM)
+		ReflectDamage(M, M?.dna?.species?.attack_type, M?.dna?.species?.punchdamagehigh)
+	if(ishuman(M))
+		TryAttachTalisman(M)
+
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/attack_paw(mob/living/carbon/human/M)
+	..()
+	if(damage_reflection && M.a_intent != INTENT_HELP)
+		ReflectDamage(M, M?.dna?.species?.attack_type, 5)
+	if(ishuman(M))
+		TryAttachTalisman(M)
+
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/attack_animal(mob/living/simple_animal/M)
+	. = ..()
+	if(!damage_reflection)
+		return
+	if(.)
+		var/damage = rand(M.melee_damage_lower, M.melee_damage_upper)
+		if(damage > 0)
+			ReflectDamage(M, M.melee_damage_type, damage)
+
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/bullet_act(obj/projectile/Proj, def_zone, piercing_hit = FALSE)
+	..()
+	if(damage_reflection && Proj.firer)
+		if(get_dist(Proj.firer, src) < 5)
+			ReflectDamage(Proj.firer, Proj.damage_type, Proj.damage)
+
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/attackby(obj/item/I, mob/living/user, params)
+	..()
+	if(ishuman(user))
+		TryAttachTalisman(user)
+	if(!damage_reflection)
+		return
+	var/damage = I.force
+	if(ishuman(user))
+		damage *= 1 + (get_attribute_level(user, JUSTICE_ATTRIBUTE)/100)
+	ReflectDamage(user, I.damtype, damage)
+
+//Reflect Code
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(. > 0)
+		damage_taken += .
+	if(health < 0)
+		damage_reflection = FALSE
+		return
+	if(!can_act)
+		return
+	if(damage_taken > 400 && !damage_reflection)
+		StartReflecting()
+
+/mob/living/simple_animal/hostile/abnormality/so_that_no_cry/proc/StartReflecting()
+	can_act = FALSE
+	damage_reflection = TRUE
+	damage_taken = 0
+	playsound(src, 'sound/abnormalities/so_that_no_cry/prepare.ogg', 50, TRUE, 7)
+	visible_message("<span class='warning>[src] assumes a stance!</span>")
+	icon_state = "so_that_no_cry_guard"
+	ChangeResistances(list(BRUTE = 1, RED_DAMAGE = 0, WHITE_DAMAGE = 0, BLACK_DAMAGE = 0, PALE_DAMAGE = 0))
+	SLEEP_CHECK_DEATH(10 SECONDS)
+	icon_state = icon_living
+	ChangeResistances(list(BRUTE = 1, RED_DAMAGE = 1.5, WHITE_DAMAGE = 1, BLACK_DAMAGE = 1, PALE_DAMAGE = 2))
```

Comment:
Heads up, the Brute change here and in the previous one is redundant, as this proc only changes the provided resistances. Not a real issue but I should let you know.